core/src/main/java/org/elasticsearch/index/query/FilteredQueryBuilder.java

```diff
@@ -40,6 +46,15 @@
     static final FilteredQueryBuilder PROTOTYPE = new FilteredQueryBuilder(null, null);
 
     /**
+     * A query that applies a filter to the results of a match_all query.
+     * @param filterBuilder The filter to apply on the query (Can be null)
+     * */
+    public FilteredQueryBuilder(@Nullable QueryBuilder filterBuilder) {
+        this.queryBuilder = new MatchAllQueryBuilder();
```

Comment:
Couldn't introduce a constant: MatchAllQueryBuilder has a setBoost/setQueryName, FilteredQueryBuilder in turn has a getQuery() so users could get access to the internal default query and change it's properties.

Went for a generateDefaultQuery() static method instead.